content/lectures/lec01.md

```diff
@@ -7,91 +7,121 @@ slides: https://docs.google.com/presentation/d/1FVklEogqEGn6zsp8YOpCuynUCCvhB6mX
-   - Common use case is deleting a file (deleting your profile picture)
-
-Idempotency is an important concept with HTTP, but also in DevOps more broadly. Many times when dealing with infrastructure you want to be able to retry an action, but don't want to actually perform that action multiple times if two requests happen to both succeed.
-
-## HTTP URLs
-
-A URL like `https://httpstat.us/200` has three components to it:
-  - `https`: This is the protocol (how we communicate with the host)
-  - `httpstat.us`: This is the host (the place we are communicating with)
-    - The port defaults for 80 for HTTP and 443 for HTTPS, but you can also specify with `:port` after the host.
-  - `/200`: This is the path (what resource we want to access on the host)
-
-## HTTP Headers
-
-Headers pass metadata about the request. Here are some examples:
-  - Authentication (include some header that verifies your identity)
-    - [`Authorization`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization)
-    - [`WWW-Authenticate`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/WWW-Authenticate)
-  - Caching (we can have a header that saves the age of the request for caching purposes)
-    - [`Cache-Control`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control)
-    - [`ETag`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag)
-  - Cookies (Browser information that is passed to the HTTP server as a header)
-    - [`Cookie`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cookie)
-  - Body info (is it in XML, JSON, or something else?)
-    - [`Content-Type`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type)
-    - [`Content-Length`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Length)
-
-## Body
-
-The body is used to pass chunks of data along with the requests. Pretty straighforward, is generally text that's given in a format specified by the `Content-Type` header.
-
-## Example HTTP Request
+# Lecture One: Networking
+
+## 7 Layers of the OSI Model
+
+The internet is built in layers that allow us to abstract away a lot of the complexity that is inherent to networks. One common model for layers is the OSI Model (from top to bottom):
+
+- Application: End user later (HTTP, FTP, SSH, DNS)
+- Presentation: Syntax layer (SSL, SSH, IMAP, FTP, JPEG)
+- Session: Synch and send to port (APIs, sockets)
+- Transport: End-to-end connections (TCP, UDP, QUIC)
+- Network: Packets (IP, ICMP, IPSec, IGMP)
+- Data Link: Frames (Ethernet, PPP, Switch, Bridge)
+- Physical: Physical structure (Coax, Fiber, Wireless, Hubs, Repeaters)
+
+## Internet Protocol (IP)
+
+IP is the lowest level of abstraction that we will uncover in this course. You can think of an IP address as a single atomic element that lives on a network. A computer is given a IP address that consists of four octets (32 bits of data) separated by dots.
```

Comment:
I'd describe it as an identifier not a unit. The atomic unit analogy only really makes sense in the context of the lecture and diagram.